src/main/java/org/broadinstitute/consent/http/service/MetricsService.java

```diff
@@ -160,4 +176,54 @@ public String getHeaderRow(Type type) {
     }
     return election;
   }
+
+  public DatasetMetrics generateDatasetMetrics(Integer datasetId) {
+
+    DatasetMetrics metrics = new DatasetMetrics();
+
+    //get datasetDTO with properties and data use restrictions
+    Set<DatasetDTO> datasets = dataSetDAO.findDatasetDTOWithPropertiesByDatasetId(datasetId);
+    if (datasets == null || datasets.isEmpty()) {
+       throw new NotFoundException("Dataset with specified ID does not exist.");
+    }
+
+    //find dars with the given datasetId in their list of datasetIds, datasetId is a String so it can be converted to jsonb in query
+    //convert all dars into smaller objects that only contain the information needed
+    List<DataAccessRequest> dars = dataAccessRequestDAO.findAllDataAccessRequestsForDatasetMetrics(Integer.toString(datasetId));
+    List<Object> darInfo = dars.stream().map(dar -> new Object() {
+      public Timestamp updateDate = dar.getUpdateDate();
+      public String projectTitle = dar.data.getProjectTitle();
+      public String darCode = dar.data.getDarCode();
+      public String nonTechRus = dar.data.getNonTechRus();
+      public String investigator = findPI(userPropertyDAO.findResearcherPropertiesByUser(dar.userId), dar.data.getResearcher()); 
+    }).collect(Collectors.toList());
```

Comment:
Similar to previous comment, prefer a real object that captures the idea of what we want. Something like `DarMetricSummary` could work. That object should take a DAR in the constructor so it can populate itself with the same info you're using in this map call.